bin/plugin/commands/performance.js

```diff
@@ -0,0 +1,228 @@
+	performanceTestDirectory,
+	environmentDirectory,
+	branch
+) {
+	log( '>> Fetching the ' + formats.success( branch ) + ' branch' );
+	await git.checkoutRemoteBranch( environmentDirectory, branch );
+
+	log( '>> Building the ' + formats.success( branch ) + ' branch' );
+	await runShellScript(
+		'npm install && npm run build',
+		environmentDirectory
+	);
+
+	log(
+		'>> Running the test on the ' + formats.success( branch ) + ' branch'
+	);
+	const results = [];
+	for ( let i = 0; i < 3; i++ ) {
+		await runShellScript(
+			'npm run test-performance',
+			performanceTestDirectory
+		);
+		const rawResults = await readJSONFile(
+			path.join(
+				performanceTestDirectory,
+				'packages/e2e-tests/specs/performance/results.json'
+			)
+		);
+		results.push( curateResults( rawResults ) );
+	}
+
+	return {
+		load: formatTime( median( results.map( ( r ) => r.load ) ) ),
+		domcontentloaded: formatTime(
+			median( results.map( ( r ) => r.domcontentloaded ) )
+		),
+		type: formatTime( median( results.map( ( r ) => r.type ) ) ),
+		minType: formatTime( median( results.map( ( r ) => r.minType ) ) ),
+		maxType: formatTime( median( results.map( ( r ) => r.maxType ) ) ),
+		focus: formatTime( median( results.map( ( r ) => r.focus ) ) ),
+		minFocus: formatTime( median( results.map( ( r ) => r.minFocus ) ) ),
+		maxFocus: formatTime( median( results.map( ( r ) => r.maxFocus ) ) ),
+	};
+}
+
+/**
+ * Runs the performances tests on an array of branches and output the result.
+ *
+ * @param {string[]} branches Branches to compare
+ */
+async function runPerformanceTests( branches ) {
+	// The default value doesn't work because commander provides an array.
+	if ( branches.length === 0 ) {
+		branches = [ 'master' ];
+	}
+
+	log(
+		formats.title( '\n💃 Performance Tests 🕺\n\n' ),
+		'Welcome! This tool runs the performance tests on multiple branches and displays a comparison table.\n' +
+			'In order to run the tests, the tool is going to load a WordPress environment on 8888 and 8889 ports.\n' +
+			'Make sure these ports are not used before continuing.\n'
+	);
+
+	await askForConfirmation( 'Ready to go? ' );
+
+	log( '>> Cloning the repository' );
+	const performanceTestDirectory = await git.clone( config.gitRepositoryURL );
+	const environmentDirectory = path.join( os.tmpdir(), uuid() );
+	log(
+		'>> Perf Tests Directory : ' +
+			formats.success( performanceTestDirectory )
+	);
+	log(
+		'>> Environment Directory : ' + formats.success( environmentDirectory )
+	);
+
+	log( '>> Installing dependencies' );
+	// The build packages is necessary for the performance folder
+	await runShellScript(
+		'npm install && npm run build:packages',
+		performanceTestDirectory
+	);
+	await runShellScript(
+		'cp -aR ' + performanceTestDirectory + ' ' + environmentDirectory
+	);
+
+	log( '>> Starting the WordPress environment' );
+	await runShellScript( 'npm run wp-env start' );
+
+	/**
+	 * @type {Object.<string, WPFormattedPerformanceResults>}
+	 */
```

Comment:
Personally I think it's fine to conserve the space and single-line when doing this sort of simple type assignment (similarly for "import-style" `@typedef`). It's occasionally unavoidable in the case of "type-casting".

```suggestion
	/** @type {Record<string, WPFormattedPerformanceResults>} */
```